version: "3.8"

services:
  go:
    build:
      context: ./projects
      dockerfile: Dockerfile
    ports:
      - 80:8080
    volumes:
      - ./:/app
    networks:
      - database
    depends_on:
      postgres:
        condition: service_healthy
    environment:
      - PGUSER=${POSTGRES_USER:?user not set}
      - PGPASSWORD=${POSTGRES_PASSWORD:?database password not set}
      - PGHOST=${POSTGRES_HOST:-postgres}
      - PGPORT=${POSTGRES_PORT:-5432}
      - PGDATABASE=${POSTGRES_DB:-sgp}

  postgres:
    image: postgres:13.2-alpine
    volumes:
      - ./database/init-scripts/:/docker-entrypoint-initdb.d/
      - sgp_postgres_data:/var/lib/postgresql/data/
    env_file: .env
    networks:
      - db_monitoring
      - database
    expose:
      - 5432
    healthcheck:
      test: ["CMD-SHELL", "sh -c 'pg_isready -U ${POSTGRES_USER} -d ${POSTGRES_DB}'"]
      interval: 5s
      timeout: 3s
      retries: 3

  adminer:
    image: adminer:4.8.1
    ports:
      - 8080:8080
    networks:
      - db_monitoring

networks:
  database:
  db_monitoring:
volumes:
    sgp_postgres_data:
        external: false